examples/example-ssl.rs

```diff
@@ -31,43 +29,46 @@ mod example {
             }
         };
 
-        // ~ OpenSSL offers a variety of complex configurations. Here is an example:
-        let mut builder = SslConnectorBuilder::new(SslMethod::tls()).unwrap();
-        {
-            builder.set_cipher_list("DEFAULT").unwrap();
-            builder.set_verify(SSL_VERIFY_PEER);
+        let rustls_config = {
+            let mut config = rustls::ClientConfig::new();
+            config
+                .root_store
+                .add_server_trust_anchors(&webpki_roots::TLS_SERVER_ROOTS);
             if let (Some(ccert), Some(ckey)) = (cfg.client_cert, cfg.client_key) {
                 info!("loading cert-file={}, key-file={}", ccert, ckey);
-
-                builder
-                    .set_certificate_file(ccert, X509_FILETYPE_PEM)
-                    .unwrap();
-                builder
-                    .set_private_key_file(ckey, X509_FILETYPE_PEM)
-                    .unwrap();
-                builder.check_private_key().unwrap();
+                let certs = {
+                    let file = fs::File::open(ccert).expect("cannot open private key file");
+                    let mut reader = BufReader::new(file);
+                    rustls::internal::pemfile::certs(&mut reader)
+                        .expect("unable to read certs file")
+                };
+
+                let keys = {
+                    let file = fs::File::open(ckey).expect("cannot open private key file");
+                    let mut reader = BufReader::new(file);
+                    rustls::internal::pemfile::rsa_private_keys(&mut reader)
+                        .expect("unable to read key file")
+                };
+
+                let key = keys[0].clone();
```

Comment:
bounds check to make sure it's non empty? Oh I guess this is also just example code tho

Reply:
Lifted from the crate's examples here: https://github.com/ctz/rustls/blob/master/rustls-mio/examples/tlsclient.rs#L412
Maybe I'll add a comment linking this..